python/packages/autogen-ext/src/autogen_ext/models/_reply_chat_completion_client.py

```
@@ -0,0 +1,162 @@
+
+    Example usage:
+    ```
+    chat_completions = [
+        "Hello, how can I assist you today?",
+        "I'm happy to help with any questions you have.",
+        "Is there anything else I can assist you with?",
+    ]
+    client = ReplayChatCompletionClient(chat_completions)
+    messages = [LLMMessage(content="What can you do?")]
+    response = await client.create(messages)
+    print(response.content)  # Output: "Hello, how can I assist you today?"
+
+    async for token in client.create_stream(messages):
+        print(token, end="")  # Output: "Hello, how can I assist you today?"
+    ```
+    """
+
+    __protocol__: ChatCompletionClient
+
+    # TODO: Support FunctionCall in responses
+    # TODO: Support logprobs in Responses
+    # TODO: Support model capabilities
+
+    def __init__(
+        self,
+        chat_completions: Sequence[Union[str, CreateResult]],
+    ):
+        self.chat_completions = list(chat_completions)
+        self._cur_usage = RequestUsage(prompt_tokens=0, completion_tokens=0)
+        self._total_usage = RequestUsage(prompt_tokens=0, completion_tokens=0)
+        self.provided_message_count = len(self.chat_completions)
+        self._model_capabilities = ModelCapabilities(vision=False, function_calling=False, json_output=False)
+        self._total_available_tokens = 10000
+
+    async def create(
+        self,
+        messages: Sequence[LLMMessage],
+        tools: Sequence[Tool | ToolSchema] = [],
+        json_output: Optional[bool] = None,
+        extra_create_args: Mapping[str, Any] = {},
+        cancellation_token: Optional[CancellationToken] = None,
+    ) -> CreateResult:
+        """Return the next completion from the list."""
+        if not self.chat_completions:
+            raise ValueError("No more mock responses available")
+
+        response = self.chat_completions.pop(0)
```

In many scenarios we want to reset the clients in test to get the same sequence again. So instead we can use index rather than popping the response. 

Also what happens if the responses are exhausted? Needs to raise an informative error here.

---

1. Makes sense, in the indexing approach the index needs to be reseted too, do you think having a `.reset` approach is good?
2. In case of messages being exhausted, currently a `ValueError("No more mock responses available")` is being thrown. There are tests for it as we, for example in `test_reply_chat_completion_client()`. Are you thinking about a custom exception?

---

> Makes sense, in the indexing approach the index needs to be reseted too, do you think having a .reset approach is good?

Yes, I think a reset approach makes it easier to work with.